### PR TITLE
Changed strategic logger logic and added user editing feature

### PR DESCRIPTION
**Pull request by yeonjaepark**
*Issue #, if available:*

*Description of changes:*
Changed the logic behind scraping for new strategic contributions now using a new table storing the date of the last scrape. Also added in the functionality of adding and modifying dates to user group start dates. 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: browser/components/EditUser.tsx [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+        groups: user.groups,
+      });
+    }
+  };
+
+  getGroupOptions = async () => {
+    const groups = [];
+    for (const key in this.state.groups) {
+      if (this.state.groups.hasOwnProperty(key)) {
+        const group = (await reqJSON(`/api/strategic/groupdetails/${key}`))
```
Comment (jamesiri) [inline browser/components/EditUser.tsx]: Hyphenate groupdetails at a minimum please.

File: server/util/strategicLogger.ts [hunk 2]
```diff
@@ -43,28 +43,26 @@ function getAmazonAliasByGithub(pg, alias) {
 async function addWhitelistedContrib(pg, contrib, projId) {
   const desc = contrib.title;
   const date = contrib.created_at;
-  const alias = (await getAmazonAliasByGithub(pg, contrib.user.login))
+  const alias = (await getCompanyAliasByGithub(pg, contrib.user.login))
```
Comment (jamesiri) [inline server/util/strategicLogger.ts:46]: No need for the additional parens here.

File: server/util/strategicLogger.ts [hunk 3]
```diff
@@ -122,18 +138,28 @@ async function updateStrategicContribs(pg) {
...
+      // find all new issues
+      const issues = await fetchNewIssues(pg, owner, repo, date);
+
+      for (const issue of issues) {
+        // check that the contribution doesn't exist and it's made by a whitelisted user
+        if (
+          issue.pull_request &&
+          (await contributionExists(pg, issue.pull_request.html_url)).length ===
+            0 &&
+          (await whitelistedContrib(pg, issue, groups))
```
Comment (jamesiri) [inline server/util/strategicLogger.ts:156]: Additional parens again.

**Review by jamesiri — COMMENTED** [review]
Minor details but looks good.

File: server/util/strategicLogger.ts (hunk 2 above)
Comment (yeonjaepark) [reply server/util/strategicLogger.ts:46]: This is needed if you look at line 47, I'm accessing .company_alias from what the promise returns, so I have to await the promise and then call it:
const alias = (await getCompanyAliasByGithub(pg, contrib,user.login)).company_alias

This linter pushed the .company_alias to the next line.

File: server/util/strategicLogger.ts (hunk 3 above)
Comment (yeonjaepark) [reply server/util/strategicLogger.ts:156]: The linter keeps adding parentheses around the second one as well.

File: browser/components/EditUser.tsx (hunk 1 above)
Comment (yeonjaepark) [reply browser/components/EditUser.tsx]: Done!

**Review by jamesiri — APPROVED** [review]

**Pull request merged**